src/vs/platform/files/common/files.ts

```diff
@@ -284,22 +284,22 @@ export class FileChangesEvent {
 	}
 
 	/**
-	 * Returns true if this change event contains the provided file with the given change type. In case of
+	 * Returns true if this change event contains the provided file with the given change type (if provided). In case of
 	 * type DELETED, this method will also return true if a folder got deleted that is the parent of the
 	 * provided file path.
 	 */
-	contains(resource: URI, type: FileChangeType): boolean {
+	contains(resource: URI, type?: FileChangeType): boolean {
 		if (!resource) {
 			return false;
 		}
 
 		return this._changes.some(change => {
-			if (change.type !== type) {
+			if (!isUndefinedOrNull(type) && change.type !== type) {
```

Comment:
@sandy081 I would move this check one level up to reduce unnecessary checks in this potentially hot loop

Reply:
Done